Script/AtomicEditor/ui/resourceEditors/TextFileResourceEditorBuilder.ts

```diff
@@ -30,6 +30,6 @@ export default class TextResourceEditorBuilder extends AbstractTextResourceEdito
 
     canHandleResource(resourcePath: string) : boolean {
         var ext = Atomic.getExtension(resourcePath).toLowerCase();
-        return ext == ".txt";
+        return ext == ".txt" || ext == ".hlsl" || ext == ".xml";
```

Comment:
With the new editor services, it might be a better idea to create new editor classes for these instead of tacking them on to the generic text editor.  Maybe make one for .hlsl and one for .xml.  It's also possible to override the base .txt editor if you have a custom file that also has a .txt extension (see how the TurboBadger editor handles this.  It overrides the plain .txt resource editor)  https://github.com/AtomicGameEngine/AtomicGameEngine/blob/master/Script/AtomicEditor/ui/ResourceEditorProvider.ts#L92-L93

That way you can detect the type of file it is by extension, or path, or filename.